src/pages/Customer/Register/Register.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from "react";
+import styles from "./Register.module.scss";
+import bgSignin from "assets/images/bgSignin.png";
+import Button from "components/Button/Button";
+import InputControl from "components/InputControl/InputControl";
+import { register, checkRegisterDetails } from "api/user/register";
+import { validateEmail, validateMobile } from "utils/util";
+import VerifyOtp from "components/verifyOtp/VerifyOtp";
+import { useNavigate } from "react-router-dom";
+const Register = (props) => {
+  const [errMsg, seterrMsg] = useState({});
+  const [RegisterDetails, setRegisterDetails] = useState({});
+  const [otpPage, setOtpPage] = useState(false);
+  const [isOtpVerified, setIsOtpVerified] = useState(false);
+  const navigate = useNavigate();
+  const changeUrl = () => {
+    navigate("/login");
+  };
+  const [values, setValues] = useState({
+    fname: "",
+    lname: "",
+    email: "",
+    mobile: "",
+  });
+  const validateForm = () => {
+    const dummyMsg = {};
+    if (values.fname === "") {
+      dummyMsg.name = "Enter name";
+    }
+    if (values.lname === "") {
+      dummyMsg.lname = "Enter last name";
+    }
+    if (values.email === "") {
+      dummyMsg.email = "Enter Email";
+    } else if (!validateEmail(values.email)) {
+      dummyMsg.email = "Enter Valid Email";
+    }
+    if (values.mobile === "") {
+      dummyMsg.mobile = "Enter mobile number";
+    } else if (!validateMobile(values.mobile)) {
+      dummyMsg.mobile = "Enter valid number";
+    }
+    seterrMsg(dummyMsg);
+    if (Object.keys(dummyMsg).length !== 0) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+  const handleRegisterCustomer = () => {
+    register({
+      firstName: values.fname,
+      lastName: values.lname,
+      mobile: values.mobile,
+      isMerchant: false,
+      email: values.email,
+    }).then((res) => {
+      console.log(res);
+    });
+  };
+
+  const handleOtpVerification = (isVerified) => {
+    console.log(isVerified);
+    setIsOtpVerified(isVerified);
+    if (isOtpVerified) handleRegisterCustomer();
+    else return;
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!validateForm()) return;
+
+    setRegisterDetails({
+      firstName: values.fname,
+      lastName: values.lname,
+      mobile: values.mobile,
+      isMerchant: false,
+      email: values.email,
+    });
+
+    checkRegisterDetails(RegisterDetails).then(async (res) => {
```

Comment:
this thing is not correct. instead of setting it to state you can make a temporary object and then pass it in the api.